lib/thread.h

```diff
@@ -207,7 +204,7 @@ extern void funcname_thread_execute(struct thread_master *,
 				    debugargdef);
 #undef debugargdef
 
-extern void thread_cancel(struct thread *);
+extern void thread_cancel(struct thread **event);
```

Comment:
In the past, our pattern has been to add _and_null to the function name and allow 1 year for transition.  This is documented in https://github.com/FRRouting/frr/blob/master/doc/developer/workflow.rst#backwards-compatibility and can be seen in  commit 49f36252d60195112caada649d94cce3bb9d496e and e208c8f94392286aaf77c6d8f2a8b4d22fa3f1d7.

I'd thing this pattern should be followed and the new call should be thread_cancel_and_null

Reply:
the goal of this work is not to _add_ a new api - it's to remove (replace) the api that has led to problems. there's no point, imo, in renaming it - and then mechanically replacing it everywhere. and there's no point, really, in leaving the problematic api behind as an attractive nuisance.